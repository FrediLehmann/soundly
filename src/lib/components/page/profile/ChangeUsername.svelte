<script lang="ts" context="module">
  import { Button, Flyin, Input } from '$lib/components/atoms';
  import { supabase } from '$lib/supabase';
</script>

<script lang="ts">
  export let username: string;

  let flyin: Flyin;

  let errorMessage = '';

  const changeUsername = async () => {
    if (!username) {
      errorMessage = 'Your username cannot be empty';
      return;
    }

    errorMessage = '';

    const { data, error } = await supabase
      .from('profiles')
      .update({ user_name: username });

    if (error) {
      flyin.show(
        'Your username could not be changed. Maybe your username is already taken.',
        'error'
      );
    } else {
      flyin.show(
        `Your username was seccessfull changed to ${data[0].user_name}.`,
        'success'
      );
    }
  };
</script>

<section>
  <h2>Username</h2>
  <p>
    Your username is <b>public</b>, it will be displayed in various places on
    our platform. It is <b>unique</b>, no two user on the platform can use the
    same username. You can change your username whenever you want as long as the
    new one is not yet taken.
  </p>
  <div class="input">
    <Input
      name="username"
      label="Username"
      type="text"
      labelSrOnly={true}
      required
      error={errorMessage}
      bind:value={username} />
  </div>
  <Button on:click={changeUsername}>Change</Button>
</section>

<Flyin bind:this={flyin} />

<style>
  h2 {
    font-size: 1.2rem;
  }

  .input {
    max-width: 70%;
    margin-block-end: 1rem;
  }

  section {
    max-width: 75%;
  }
</style>
