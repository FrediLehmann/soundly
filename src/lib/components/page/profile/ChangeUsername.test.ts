/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import ChangeUsername from './ChangeUsername.svelte';

describe('<ChangeUsername /> component', () => {
  test('renders correctly', () => {
    const { component } = render(ChangeUsername);
    expect(component).toMatchSnapshot();
  });
});
