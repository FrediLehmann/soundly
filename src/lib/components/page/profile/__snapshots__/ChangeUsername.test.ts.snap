// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeUsername /> component renders correctly 1`] = `
ChangeUsername {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "",
      Flyin {
        "$$": Object {
          "after_update": Array [],
          "before_update": Array [],
          "bound": Object {},
          "callbacks": Object {},
          "context": Map {},
          "ctx": Array [
            false,
            undefined,
            undefined,
            [Function],
            [Function],
          ],
          "dirty": Array [
            -1,
          ],
          "fragment": Object {
            "c": [Function],
            "d": [Function],
            "i": [Function],
            "l": [Function],
            "m": [Function],
            "o": [Function],
            "p": [Function],
          },
          "not_equal": [Function],
          "on_destroy": Array [],
          "on_disconnect": Array [],
          "on_mount": Array [],
          "props": Object {
            "show": 4,
          },
          "root": <div>
            <section
              class="svelte-1sz96f6"
            >
              <h2
                class="svelte-1sz96f6"
              >
                Username
              </h2>
               
              <p>
                Your username is 
                <b>
                  public
                </b>
                , it will be displayed in various places on
    our platform. It is 
                <b>
                  unique
                </b>
                , no two user on the platform can use the
    same username. You can change your username whenever you want as long as the
    new one is not yet taken.
              </p>
               
              <div
                class="input svelte-1sz96f6"
              >
                <div
                  class="container svelte-1ulq3bj"
                  data-required="true"
                  data-sr-only="true"
                >
                  <label
                    class="svelte-1ulq3bj"
                    for="username"
                  >
                    Username
                  </label>
                   
                  <input
                    class="svelte-1ulq3bj"
                    id="username"
                    required=""
                    type="text"
                  />
                   
                </div>
              </div>
               
              <button
                class="svelte-1dsjpe6"
                data-type="secondary"
                type="button"
              >
                Change
              </button>
              
            </section>
             
            
          </div>,
          "skip_bound": false,
          "update": [Function],
        },
        "$capture_state": [Function],
        "$inject_state": [Function],
      },
      "",
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "username": 0,
    },
    "root": <div>
      <section
        class="svelte-1sz96f6"
      >
        <h2
          class="svelte-1sz96f6"
        >
          Username
        </h2>
         
        <p>
          Your username is 
          <b>
            public
          </b>
          , it will be displayed in various places on
    our platform. It is 
          <b>
            unique
          </b>
          , no two user on the platform can use the
    same username. You can change your username whenever you want as long as the
    new one is not yet taken.
        </p>
         
        <div
          class="input svelte-1sz96f6"
        >
          <div
            class="container svelte-1ulq3bj"
            data-required="true"
            data-sr-only="true"
          >
            <label
              class="svelte-1ulq3bj"
              for="username"
            >
              Username
            </label>
             
            <input
              class="svelte-1ulq3bj"
              id="username"
              required=""
              type="text"
            />
             
          </div>
        </div>
         
        <button
          class="svelte-1dsjpe6"
          data-type="secondary"
          type="button"
        >
          Change
        </button>
        
      </section>
       
      
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
